### PR TITLE
Travis: Upload builds to GitHub Releases

### PR DESCRIPTION
**Pull request by JayFoxRox**
Alternative to #10 
The version name will be the rather ugly "untagged-<commit>" but it gets the job done.

---

Uploads tar.xz to GitHub Releases. I consider switching to zip for ease-of-use but want feedback on that.

This needs some changes still for the citra-emu OAUTH. @bunnei 
(I've generated my token using travis.rb as recommended on the Travis docs)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +15,20 @@ install:
...
+
+deploy:
+  provider: releases
+  api_key:
+    secure: MvtH+Xy9GWy7h/rYxjXQ0hMC+6EeXsnULE72JZrnw1TxHWlIotkQOMj4WgyWrI0l8a3UIkIsY75STBe4xlTNaS2tM5Zg3PsWHisEikSkTbw1lO7FDygdklOvZAsUJ14DDBer0JE1ME81kKfQePaQggrWJdTIsgs16WPUuremoJ2vjHZ16/rXlEwUVs4mKD9CCF17kul0ZbiQYbqkX/TQIo5o1iz9kVjFHvIfuGNsITNPrBXUwZVgS3W+XhkIfdW+KpRNrt1tQ46t/TFB26wMfqe2kZuh2mdo+Vav5oFkE7qJQx0qPwKm42WuVAOVpV7XqJW8Sj1g9lVvQlzek+2jkkkGI3HPFfeeRXabe+0uIRKCVzh1B0E7jx6eh7M4qi9ePaPCU0QvUucjCDi7bUJgTKDBnUgSNLgWwnJZcjmQZjrV6GfwfpRsu8Y3CULOwrSUfWZS9QPy4foOqTejK7QBIbLbAzjfOHvdr0wt9BbGbrQVNJbsbLElNa8+Qx8fcExswLvO/Q2e8RC+f3FtoUs8g8hNghZyM4v28B6FzyELtqo09wK7MzueHMpdJDnifAf03/Soy6qCrUHd1EnGCm2BnwdefJKqvJuD9de/nqpHmSsEC7CZaBMWwnxcyu5H7RU2s1kAWMP3ndr4RxiiPSklXo0IXTaV9RyaUcKSGwNRZkc=
+  skip_cleanup: true
+  file_glob: true
+  file: "${ARCHIVE_NAME}"
+  on:
+    repo: JayFoxRox/3dsutils
```
Comment (JayFoxRox) [inline .travis.yml]: Needs repo update

File: .travis.yml [hunk 2]
```diff
@@ -11,6 +15,20 @@ install:
...
 
 script:
   - make
+
+after_success: source ./.travis-package.sh
+
+deploy:
+  provider: releases
+  api_key:
+    secure: MvtH+Xy9GWy7h/rYxjXQ0hMC+6EeXsnULE72JZrnw1TxHWlIotkQOMj4WgyWrI0l8a3UIkIsY75STBe4xlTNaS2tM5Zg3PsWHisEikSkTbw1lO7FDygdklOvZAsUJ14DDBer0JE1ME81kKfQePaQggrWJdTIsgs16WPUuremoJ2vjHZ16/rXlEwUVs4mKD9CCF17kul0ZbiQYbqkX/TQIo5o1iz9kVjFHvIfuGNsITNPrBXUwZVgS3W+XhkIfdW+KpRNrt1tQ46t/TFB26wMfqe2kZuh2mdo+Vav5oFkE7qJQx0qPwKm42WuVAOVpV7XqJW8Sj1g9lVvQlzek+2jkkkGI3HPFfeeRXabe+0uIRKCVzh1B0E7jx6eh7M4qi9ePaPCU0QvUucjCDi7bUJgTKDBnUgSNLgWwnJZcjmQZjrV6GfwfpRsu8Y3CULOwrSUfWZS9QPy4foOqTejK7QBIbLbAzjfOHvdr0wt9BbGbrQVNJbsbLElNa8+Qx8fcExswLvO/Q2e8RC+f3FtoUs8g8hNghZyM4v28B6FzyELtqo09wK7MzueHMpdJDnifAf03/Soy6qCrUHd1EnGCm2BnwdefJKqvJuD9de/nqpHmSsEC7CZaBMWwnxcyu5H7RU2s1kAWMP3ndr4RxiiPSklXo0IXTaV9RyaUcKSGwNRZkc=
```
Comment (JayFoxRox) [inline .travis.yml]: needs OAUTH update

File: .travis-package.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    cp 3dsutils.3dsx "$REV_NAME"
+    cp 3dsutils.elf "$REV_NAME"
+    cp send-exec.py "$REV_NAME"
+
+    # Copy documentation
+    cp license.txt "$REV_NAME"
+    cp README.md "$REV_NAME"
+
+    export ARCHIVE_NAME="${REV_NAME}.tar.xz"
+    tar -cJvf "$ARCHIVE_NAME" "$REV_NAME"
```
Comment (JayFoxRox) [inline .travis-package.sh]: Shall I use zip instead?

Comment (archshift) [inline .travis-package.sh]: I used xz because of its superior compression, but it's not necessary if github will be hosting the files. Zip works!

**Comment by JayFoxRox** [comment]
I consider this PR failed.

Closed.

**Pull request closed without merging**